app/services/importers/okc_blotter/pdf.rb

```diff
@@ -0,0 +1,137 @@
+        )
+        @json = JSON.parse(response.body)
+        return @json if response.success?
+
+        raise StandardError, "Error parsing pdf: #{@json['message']}"
+      end
+
+      def pdf
+        return @pdf if @pdf
+
+        pdf = begin
+          Bucket.new.get_object("#{s3_path}/#{@date}.pdf").body
+        rescue StandardError
+          false
+        end
+        @pdf = pdf if pdf
+        pdf
+      end
+
+      def build_bookings(booking_hash)
+        ::OkcBlotter::Booking.new({
+                                    dob: booking_hash['dob'],
+                                    sex: booking_hash['sex'],
+                                    last_name: booking_hash['lastName'],
+                                    first_name: booking_hash['firstName'],
+                                    race: booking_hash['race'],
+                                    booking_number: booking_hash['bookingNumber'],
+                                    booking_date: booking_hash['bookingDate'],
+                                    inmate_number: booking_hash['inmateNumber'],
+                                    transient: booking_hash['transient'],
+                                    zip: booking_hash['zip'],
+                                    booking_type: booking_hash['bookingType'],
+                                    offenses: booking_hash['offenses'].map do |offense_hash|
+                                      build_offense(offense_hash)
+                                    end
+                                  })
+      end
+
+      def build_offense(offense_hash)
+        ::OkcBlotter::Offense.new({
+                                    type: offense_hash['type'],
+                                    bond: offense_hash['bond'],
+                                    code: offense_hash['code'],
+                                    dispo: offense_hash['dispo'],
+                                    charge: offense_hash['charge'],
+                                    warrant_number: offense_hash['warrantNumber'],
+                                    citation_number: offense_hash['citationNumber']
+                                  })
+      end
+
+      def s3_path
+        'okc_blotter'
+      end
+
+      def url
+        'https://okc-blotter.herokuapp.com'
+      end
+
+      def auth_token
+        ENV.fetch('OKC_BLOTTER_AUTH_TOKEN', nil)
+      end
+
+      def self.import_since_last_run
+        last_run_date = ::OkcBlotter::Pdf.order(date: :desc).first.date
+        (last_run_date + 1.day..DateTime.now.to_date).each do |date|
+          perform(date)
+        rescue StandardError => e
+          puts "error processing #{date}. Data not saved" # TODO: log these somewhere?
+          puts "error was: #{e}"
+        end
+      end
+
+      # NOTE: PDFs only are on the website for 30 days but if pdfs are saved to s3 you may be able to parse longer
+      def self.import_missing_dates(since_date = (DateTime.now - 1.month).to_date)
```

Comment:
seems to collect dates and run the perform method on them, but from what I saw the perform method doesn't have a date parameter